### DISCUSSION TITLE
Работа с regex-custom

### DISCUSSION THREAD

**Discussion by RazGame** [xtrime-ru/antizapret-vpn-docker#97] [Q&A] [answered]
Можете подсказать, как правильно включать сайты в список ?
Вот кусок из текущего моего файла

microsoft\.[^.]*$
windowsupdate\.[^.]*$
windows\.[^.]*$
bing\.[^.]*$

и я вот не совсем понимаю, правильно ли я его сконфигурировал.
цель - проксировать указанные ресурсы, но как работает парсер, он построчно проверяет из файла ? или нужна одна регулярка на все сайты?

**Comment by xtrime-ru** [comment, marked as answer]
Вот парсер: https://github.com/xtrime-ru/antizapret-vpn-docker/blob/master/rootfs/root/antizapret/build_regex.sh

Он склеивает все строки в одну регулярку вида /(microsoft.[^.]$|windowsupdate.[^.]$|...)/

Т.е. формат у вас верный. 

После того как отработает doall можно проверить правила так: 
```
docker exec -it antizapret-vpn tail -n5  /etc/knot-resolver/knot-aliases-alt.conf
```
